### DISCUSSION TITLE
[Bug]: Using Ionic Toggle Component With Controlled Input

### DISCUSSION THREAD

**Discussion by aaronksaunders** [react-hook-form/react-hook-form#6231] [Q&A]
### Version Number

7

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-js-forked-5nshs?file=/src/App.js

### Steps to reproduce

Click the toggle and the value is not changing appropriately.

ionic has a separate handler for the change event and it returns its value differently, I cannot figure out how to set the default value and get the appropriate value returned when toggling the control.

### Expected behaviour

get the appropriate value from the control and having the ability to set the default value

### What browsers are you seeing the problem on?

Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
https://ionicframework.com/docs/api/toggle

follow the doc, the event is not standard `e.target.value` or `e.target.checked`, but `e.detail.checked`

`return <IonToggle onIonChange={e => field.onChange(e.detail.checked)} />;`

<img width="800" alt="Screen Shot 2021-08-14 at 9 06 15 am" src="https://user-images.githubusercontent.com/10513364/129425926-98f6b8f5-e500-4ef7-9425-7c329834e1c3.png">
